modules/gapi/test/gapi_transform_tests.cpp

```diff
@@ -0,0 +1,148 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+#include <tuple>
+
+#include "test_precomp.hpp"
+#include "opencv2/gapi/gtransform.hpp"
+
+namespace opencv_test
+{
+
+namespace
+{
+using GMat2 = std::tuple<GMat, GMat>;
+using GMat3 = std::tuple<GMat, GMat, GMat>;
+using GMat = cv::GMat;
+
+GAPI_TRANSFORM(my_transform, <GMat(GMat, GMat)>, "does nothing")
+{
+    static GMat pattern(GMat, GMat)
+    {
+        return {};
+    };
+
+    static GMat substitute(GMat, GMat)
+    {
+        return {};
+    }
+};
+
+GAPI_TRANSFORM(another_transform, <GMat3(GMat, GMat)>, "does nothing")
+{
+    static GMat3 pattern(GMat, GMat)
+    {
+        return {};
+    };
+
+    static GMat3 substitute(GMat, GMat)
+    {
+        return {};
+    }
+};
+
+GAPI_TRANSFORM(copy_transform, <GMat3(GMat, GMat)>, "does nothing")
+{
+    static GMat3 pattern(GMat, GMat)
+    {
+        return {};
+    };
+
+    static GMat3 substitute(GMat, GMat)
+    {
+        return {};
+    }
+};
+
+GAPI_TRANSFORM(simple_transform, <GMat(GMat)>, "does nothing")
+{
+    static GMat pattern(GMat)
+    {
+        return {};
+    };
+
+    static GMat substitute(GMat)
+    {
+        return {};
+    }
+};
+
+GAPI_TRANSFORM(another_simple_transform, <GMat2(GMat)>, "does nothing too")
+{
+    static GMat2 pattern(GMat)
+    {
+        return {};
+    };
+
+    static GMat2 substitute(GMat)
+    {
+        return {};
+    }
+};
```

Comment:
`GMat`s are nice, what about `GScalar`s, `GArray<T>`s? :)

Reply:
Will add

Reply:
And so it worked? :)

Reply:
Ah I see, for the output type. OK